### PR TITLE
Doc Clean Up Cloud Provider Access

### PR DESCRIPTION
**Pull request by Zuhairahmed**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.
Clean up Cloud Provider Access resource docs to make more clear that two options are either-or/mutually exclusive as well link our example directly into definitions  up top. 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
         with:
           go-version: 1.18
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.3.1
+        uses: golangci/golangci-lint-action@v3.4.0
```
Comment (themantissa) [inline .github/workflows/golangci-lint.yml:19]: You are somehow picking up changes not in your PR.  Make sure you are submitting what you intend.

**Review by themantissa — CHANGES_REQUESTED** [review]
The doc update is good but you've picked up a linter version change.  Make sure you are using an updated branch so your PR is specific to what you are working on.

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (Zuhairahmed) [reply .github/workflows/golangci-lint.yml:19]: i think this is from a dependabot that's already merged into master, but not part of this v1.8.1-staging branch so gets pulled in automatically. @martinstibbe do you know if there is any easy way to remove these from future PRs? or if not, any concerns if we leave as is?

Comment (martinstibbe) [reply .github/workflows/golangci-lint.yml:19]: I have rebased 1.8.1 staging with master so lint version should match up.

Comment (Zuhairahmed) [reply .github/workflows/golangci-lint.yml:19]: Thanks!

**Review by themantissa — APPROVED** [review]
LGTM!

**Pull request merged**